### ISSUE TITLE
Exposing some of `templates.Filename` parameters in `templates.Path` and `as_path()`, `as_filename()`

### ISSUE THREAD

**Issue by Jmennius** [beetbox/confuse#162]
`templates.Filename` supports configuring how the path is resolved (relative to config, some directory and so on).
I believe it would be nice to have some (or all) of those exposed for `as_path()` and `as_filename()` methods.
I am willing to implement this, just want to make sure this is something that would be accepted :)

**Comment by sampsyo** [comment]
Yes, sounds good to me!! Thanks for being willing to take this on.
